homeassistant/helpers/translation.py

```diff
@@ -271,71 +289,22 @@ async def async_get_translations(
     Otherwise default to loaded intgrations combined with config flow
     integrations if config_flow is true.
     """
-    lock = hass.data.get(TRANSLATION_LOAD_LOCK)
-    if lock is None:
-        lock = hass.data[TRANSLATION_LOAD_LOCK] = asyncio.Lock()
+    lock = hass.data.setdefault(TRANSLATION_LOAD_LOCK, asyncio.Lock())
 
     if integration is not None:
         components = {integration}
     elif config_flow:
-        # When it's a config flow, we're going to merge the cached loaded component results
-        # with the integrations that have not been loaded yet. We merge this at the end.
-        # We can't cache with config flow, as we can't monitor it during runtime.
         components = (await async_get_config_flows(hass)) - hass.config.components
-    else:
+    elif category == "state":
         # Only 'state' supports merging, so remove platforms from selection
```

Comment:
This comment should be moved to the `else` block. (it was incorrectly placed before too)

Reply:
fixed